src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -772,27 +806,51 @@ end subroutine change_thickness_using_melt
 
 !> This subroutine adds the mechanical forcing fields and perhaps shelf areas, based on
 !! the ice state in ice_shelf_CS.
-subroutine add_shelf_forces(G, US, CS, forces, do_shelf_area)
-  type(ocean_grid_type), intent(inout) :: G    !< The ocean's grid structure.
+subroutine add_shelf_forces(Ocn_grid, US, CS, forces_in, do_shelf_area, external_call)
+  type(ocean_grid_type), intent(in)    :: Ocn_grid !< The ocean's grid structure.
   type(unit_scale_type), intent(in)    :: US   !< A dimensional unit scaling type
   type(ice_shelf_CS),    pointer       :: CS   !< This module's control structure.
-  type(mech_forcing),    intent(inout) :: forces !< A structure with the driving mechanical forces
+  type(mech_forcing),    target        :: forces_in !< A structure with the driving mechanical forces
```

Comment:
There should be an intent declaration for the forces_in structure, and similarly for the fluxes_in structure above at line 213.